insights-notification-schemas-java/src/main/java/com/redhat/cloud/notifications/ingress/Decoder.java

```diff
@@ -0,0 +1,79 @@
+package com.redhat.cloud.notifications.ingress;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.JsonDecoder;
+import org.apache.avro.specific.SpecificData;
+import org.apache.avro.specific.SpecificDatumReader;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Optional;
+
+public class Decoder {
+
+    private static final String VERSION_PROPERTY = "version";
+    private static final String DEFAULT_VERSION = "v1.0.0";
+
+    private final Registry registry;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public Decoder(Registry registry) {
+        this.registry = registry;
+    }
```

Comment:
The application using it will be in charge of it.
see: https://github.com/RedHatInsights/notifications-backend/pull/595/files#diff-f822c278f715da24f20cfb937297e6d9646a3c5182d89e0a11fefbd40c53ac9bR13-R14